MiscVBAFunctions/Modules/MiscNewKeys.bas

```diff
@@ -13,11 +13,11 @@ Public Function NewSheetName(Name As String, Optional WB As Workbook)
     NewSheetName = Left(Name, 31)
 
     ' If Not Fn.hasKey(WB.Sheets, NewSheetName) Then
-    If Not hasKey(WB.Sheets, NewSheetName) Then
+    If Not hasKey(WB.Worksheets, NewSheetName) Then
```

Comment:
This is a tricky one, but in this instance, we should actually use `Sheet`, as a "Chart" sheet and "Worksheet" are not allowed to have the same name.
For example, if you have a "Chart" sheet called "foo", you cannot have a "Worksheet" called "foo", it needs to resolve to "foo1".

Reply:
I had a quick look and all the other instances seems to be correct to use `Worksheets` as opposed to `Sheets` @BartJan142